framework/test/driver/ServiceRegistryTest.cpp

```diff
@@ -80,6 +83,32 @@ void TestMultipleServiceRegistrations(BundleContext context)
   US_TEST_CONDITION_REQUIRED(!ref, "Testing for invalid service reference")
 }
 
+// Spawned thread repeatedly calls the IsConvertibleTo method and when the main thread
+// unregisters the service, it used to result in a crash.
+void TestUnregisterFix(BundleContext context)
```

Comment:
This function should be guarded by a `#ifdef US_ENABLE_THREADING_SUPPORT`. One of the Travis test failures is related to the code executing without the synchronization primitives "enabled".

Reply:
Thanks Sascha. I guarded this test with a US_ENABLE_THREADING_SUPPORT and in the `#else` part, I have the single-threaded test. (And yes, the bug manifested before the fix and not after the fix.)